PoGo.NecroBot.Logic/Tasks/CatchPokemonTask.cs

```diff
@@ -433,6 +442,51 @@ private static bool CatchThresholdExceeds(ISession session, CancellationToken ca
             }
         }
 
+        private static bool IsNotMetWithCatchCriteria(ISession session, PokemonData encounteredPokemon, double pokemonIv, double lv, int? cp)
+        {
+            if (session.LogicSettings.UsePokemonToNotCatchFilter && session.LogicSettings.PokemonsNotToCatch.Contains(encounteredPokemon.PokemonId)) return true;
+            if(session.LogicSettings.UseTransferFilterToCatch && session.LogicSettings.PokemonsTransferFilter.ContainsKey(encounteredPokemon.PokemonId))
+            {
+                var filter = session.LogicSettings.PokemonsTransferFilter[encounteredPokemon.PokemonId];
+                if(filter != null && filter.CatchOnlyPokemonMeetTransferCriteria)
+                {
+                    if(filter.KeepMinOperator =="and" 
+                        && ((cp.HasValue && cp.Value < filter.KeepMinCp)  
+                        || pokemonIv < filter.KeepMinIvPercentage 
+                        || (filter.UseKeepMinLvl && lv < filter.KeepMinLvl)))
+                    {
+                        return true;//not catch pokemon
+                    }
+
+                    if (filter.KeepMinOperator == "or" && ((!cp.HasValue || cp < filter.KeepMinCp) 
+                        && pokemonIv < filter.KeepMinIvPercentage 
+                        && (!filter.UseKeepMinLvl || lv < filter.KeepMinLvl)))
+                    {
+                        return true;//not catch pokemon
+                    }
+                    //check for move
+
+                    var move1 = PokemonInfo.GetPokemonMove1(encounteredPokemon).ToString();
+                    var move2 = PokemonInfo.GetPokemonMove2(encounteredPokemon).ToString();
+
+                    if (filter.MovesOperator == "or" && (filter.Moves.Count ==0 || filter.Moves.Any(x=>x[0] == encounteredPokemon.Move1 || x[1] == encounteredPokemon.Move2)))
```

Comment:
Ohhh, This part needs revision.
I see what happened. 
The MovesOperator is **not** for Move 1 and/or Move 2.
The "MovesOperator": "or/and", means that it will transfer, pokemon if the MOVES **(AND)/(OR)** CP/LVL/IV are above threshold.

EXAMPLE:
"Golduck": {
      "KeepMinCp": 1000,
      "KeepMinIvPercentage": 90.0,
      "KeepMinLvl": 21,
      "UseKeepMinLvl": true,
      "KeepMinOperator": "or",
      "KeepMinDuplicatePokemon": 0,
      "Moves": [
        [
          "waterGunFast",
          "hydroPump"
        ]
      ],
      "DeprecatedMoves": null,
      "MovesOperator": "CHECK BELOW",

So, If 
"MovesOperator": "or"
CP:1000, IV:90, LVL:21 Moves:"confusionFast","hydroPump" --> Kept
CP:10, IV:10, LVL:10 Moves:"waterGunFast","hydroPump" --> Kept
CP:10, IV:10, LVL:21 Moves:"confusionFast","hydroPump" --> Transferred


"MovesOperator": "and"
CP:1000, IV:90, LVL:21 Moves:"confusionFast","hydroPump" --> Transferred
CP:10, IV:10, LVL:10 Moves:"waterGunFast","hydroPump" --> Transferred
CP:1000, IV:90, LVL:21 Moves:"waterGunFast","hydroPump" --> Kept


Moves are paired together and having only one good move is not good. Have to have the move"set" or pair of both moves. That's how the filter is created and it checks both moves together, when MovesOperator checks CP/IV/LVL **and/or** Move(1&2).
;-)